openmct/ait_integration.js

```diff
@@ -0,0 +1,338 @@
+
+// Prints debug message to console, only if debug is enabled
+function debugMsg(msg) {
+    if (ait_debug)
+        console.log("OpenMct/AIT: "+msg);
+}
+
+
+//---------------------------------------------
+
+//Sets config, creates providers for Objects and Composition
+//for AIT
+function AITIntegration(config) {
+
+
+    ait_host  = AIT_HOST_DEFAULT;
+    ait_post  = AIT_PORT_DEFAULT;
+    ait_debug = DEBUG_ENABLED_DEFAULT;
+
+    //check for configuration overrides
+    if (config != null)
+    {
+        if (config.hasOwnProperty('debug'))
+        {
+            ait_debug = (config.debug === "true");
+        }
+        if (config.hasOwnProperty('host'))
+        {
+            ait_host = config.host;
+
+        }
+        if (config.hasOwnProperty('port'))
+        {
+            //Really Javascript? Gotta jump through THIS hoop??
+            ait_port = (Number.isInteger(config.port)) ? config.port :
+                       (parseInt(config.port) || AIT_PORT_DEFAULT);
+        }
+    }
+
+    //You will only see these if DEBUG was enabled
+    debugMsg("AIT Debug set to: "+ait_debug);
+    debugMsg("AIT Host  set to: "+ait_host);
+    debugMsg("AIT Port  set to: "+ait_port);
+
+    tlmdictPromise  = getDictionary(ait_host, ait_port);
+
+    let objectProvider = {
+        get: function (identifier) {
+
+            return tlmdictPromise.then(function (dictionary) {
+                if (identifier.key === 'spacecraft') {
+                    return {
+                        identifier: identifier,
+                        name: dictionary.name,
+                        type: 'folder',
+                        location: 'ROOT'
+                    };
+                } else {
+                    let measurement = dictionary.measurements.filter(function (m) {
+                        return m.key === identifier.key;
+                    })[0];
+                    return {
+                        identifier: identifier,
+                        name: measurement.name,
+                        type: 'telemetry',
+                        telemetry: {
+                            values: measurement.values
+                        },
+                        location: 'taxonomy:spacecraft'
+                    };
+                }
+            });
+        }
+    };
+
+    let compositionProvider = {
+        appliesTo: function (domainObject) {
+            return domainObject.identifier.namespace === 'taxonomy' &&
+                   domainObject.type === 'folder';
+        },
+        load: function (domainObject) {
+            return tlmdictPromise
+                .then(function (dictionary) {
+                    return dictionary.measurements.map(function (m) {
+                        return {
+                            namespace: 'taxonomy',
+                            key: m.key
+                        };
+                    });
+                });
+        } //,
+        // loadHeirarchy: function (domainObject) {
```

Comment:
Are we keeping this around for something or did it just not get cleaned out?

Reply:
@MJJoyce 1) I added OpenMCT to Sphinx docs, 2) Decided to leave that commented out code in there for now as we do plan to support treating a high-level Packet as folders with telem entries within.  Hopefully the squash-commit worked.  🤞